Assets/Project/Scripts/UnityChanControlScript.cs

```diff
@@ -54,178 +63,114 @@ private void Start()
             orgColHight = col.height;
             orgVectColCenter = col.center;
 
-            // populate dictionary
-            animationStates = new Dictionary<int, Action>();
-            animationStates.Add(idleState, () => WhenIdle());
-            animationStates.Add(locoState, () => WhenRunning());
-            animationStates.Add(jumpState, () => WhenJumping());
-            animationStates.Add(restState, () => WhenResting());
-            animationStates.Add(slideState, () => WhenSliding());
-            animationStates.Add(attackState, () => WhenAttacking());
+            // Create Abilities, by default is disabled      
+            var jumpingAbility = new Ability() { AnimParamName = jumpParam, IsEnabled = false ,Key = KeyCode.Space, FuncToCall = () => JumpControls()};
+            var slidinAbility = new Ability() { AnimParamName = slideParam, IsEnabled = false ,Key = KeyCode.S, FuncToCall = () => SlidingControls()};
+            var attackAbility = new Ability() { AnimParamName = attackParam, IsEnabled = false ,Key = KeyCode.Mouse0, FuncToCall = () => AttackControls()};
+            var abilityToDie = new Ability() { AnimParamName = deathParam, IsEnabled = false, Key = KeyCode.K, FuncToCall = () => DeathControls() };
```

Comment:
Don't think lambda expression is necessary here. Can do 
`FuncToCall = DeathControls`

Reply:
the lambda is to call that function so you don't  have to invoke it later just call the dictionary